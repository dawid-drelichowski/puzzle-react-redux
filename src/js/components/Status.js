import React from 'react';

export default class extends React.Component {
  static propTypes = {
    piecesLeftCount: React.PropTypes.number.isRequired
  };
  render() {
    let status = <div className="alert label-warning text-white">Left: <span className="badge">{this.props.piecesLeftCount}</span></div>;

    if (this.props.piecesLeftCount === 0) {
      status = <div className="alert label-success text-white">âœ” Done!</div>;
    }
    return <div className="status col-lg-2">
      {status}
    </div>;
  }
}