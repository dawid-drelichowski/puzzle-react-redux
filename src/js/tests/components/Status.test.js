import React from 'react';
import Status from 'components/Status';
import renderer from 'react-test-renderer';

describe('Status component', () => {
  function getComponent(piecesLeftCount) {
    return renderer.create(
      <Status piecesLeftCount={piecesLeftCount} />
    );
  }

  it('should show pieces left', () => {
    expect(getComponent(9).toJSON()).toMatchSnapshot();
  });

  it('should show "Done" when no pieces left', () => {
    expect(getComponent(0).toJSON()).toMatchSnapshot();
  });
});
