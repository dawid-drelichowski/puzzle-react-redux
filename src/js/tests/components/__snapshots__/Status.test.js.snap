exports[`Status component should show "Done" when no pieces left 1`] = `
<div
  className="status col-lg-2">
  <div
    className="alert label-success text-white">
    âœ” Done!
  </div>
</div>
`;

exports[`Status component should show pieces left 1`] = `
<div
  className="status col-lg-2">
  <div
    className="alert label-warning text-white">
    Left: 
    <span
      className="badge">
      9
    </span>
  </div>
</div>
`;
